dask/dataframe/multi.py

```diff
@@ -307,6 +312,241 @@ def merge_indexed_dataframes(lhs, rhs, left_index=True, right_index=True, **kwar
 shuffle_func = shuffle  # name sometimes conflicts with keyword argument
 
 
+def _bcast_heuristic(n_l, n_r):
+    """Simple heuristic to decide if a broadcast join is likely
+    to outperform a shuffle-based merge.
+    """
+    ntasks_shuffle = n_r * math.log2(n_r) + n_l * math.log2(n_l)
+    ntasks_bcast = n_r * n_l + n_l
+    return ntasks_bcast < ntasks_shuffle * 0.75
```

Comment:
I have a slight preference to include this logic directly at the call site if you're ok with that.  [Rationale here](https://matthewrocklin.com/blog/work/2019/06/23/avoid-indirection)

Reply:
I'm curious, why the 0.75 factor?

Reply:
> I have a slight preference to include this logic directly at the call site if you're ok with that. Rationale here

Sound good to me.

>I'm curious, why the 0.75 factor?

Just a rough scaling factor (I noticed it was best to prefer "shuffle" when the heuristic calculation was "close").  The heuristic is absolutely something I need to revisit/test :)

Reply:
Should this factor be configurable perhaps with a default of 0.75? May allow us to adjust it more easily in the future or allow informed users to adjust based on their needs

Reply:
>Should this factor be configurable perhaps with a default of 0.75? May allow us to adjust it more easily in the future or allow informed users to adjust based on their needs

Yes - I am thinking that the user should have the option to congure the heuristic somehow, but I'm not sure if it should be this scaling factor.  If we decide that the existing heuristic actually makes sense, then this may indeed be the best approach.